### PR TITLE
Add withoutSizeLimit overrides for Scala API

### LINKED ISSUE #19983: Scala API withoutSizeLimit overrides are missing
effectively switching from Scala DSL to Java DSL when `withoutSizeLimit` is used.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #19983

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 1]
```diff
@@ -228,7 +243,7 @@ object HttpEntity {
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
     override def withSizeLimit(maxBytes: Long): UniversalEntity =
-      if (data.length <= maxBytes) this
+      if (data.length <= maxBytes || isKnownEmpty) this
```
Comment (2m) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: @ktoso this is something that I added to make `HttpEntity.Empty.withoutSizeLimit` run without `require` exceptions.

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: Looks good :+1:

**Comment by ktoso** [comment]
LGTM, thanks!

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 2]
```diff
@@ -126,6 +126,11 @@ sealed trait RequestEntity extends HttpEntity with jm.RequestEntity with Respons
    */
   def withSizeLimit(maxBytes: Long): RequestEntity
 
+  /**
+   * See [[HttpEntity#withoutSizeLimit]].
+   */
+  def withoutSizeLimit: RequestEntity
```
Comment (patriknw) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: add `override` on this and other methods?

**Comment by patriknw** [comment]
LGTM

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala (hunk 2 above)
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: We can attempt that as a sweeping change across the http codebase I think. It does not use override, we had some discusions on that, moving towards using it more now.

**Pull request merged**